File: ProRent/Domain/Filters/RealEstateFilter.cs
using ProRent.Domain.Enums;

namespace ProRent.Domain.Filters
{
    public class RealEstateFilter
    {
        public RealEstateType? Type { get; set; }

        public string Name { get; set; }

        public string Address { get; set; }

        public double? MaxArea { get; set; }

        public double? MinArea { get; set; }

        public double? maxRent { get; set; }

        public double? minRent { get; set; }
    }
}

File: client/src/app/shared/models/RealEstate/real-estate-filter.ts
import { RealEstateType } from "../../enums/real-estate-type.enum";

export class RealEstateFilter {
  type?: RealEstateType;
  name?: string;
  address?: string;
  maxArea?: number;
  minArea?: number;
  maxRentValue?: number;
  minRentValue?: number;

  public set AreaRange(filter: { max?: number; min?: number }) {
    this.maxArea = filter?.max;
    this.minArea = filter?.min;
  }

  public set RentRange(filter: { max?: number; min?: number }) {
    this.maxRentValue = filter?.max;
    this.minRentValue = filter?.min;
  }
}
